lib/authentication_methods.rb

```diff
@@ -246,6 +245,31 @@ def require_user
   end
   protected :require_user
 
+  def require_user_render_json_unauthorized
```

Comment:
Note: I could have continued to use `require_user` and just changed the call from JavaScript into `set_user_retained_data` to only accept JSON responses instead of HTML, but I wanted to add the redirect URL into the response so I introduced this new method.

Reply:
That makes sense to me. I always prefer more verbose/detailed responses that enables more options to be performed.